### PR TITLE
Use grid store instead of getGrid

### PR DESCRIPTION
**Pull request by zaelgohary**
### Changes

- Use grid store instead of getGrid in all Dashboard pages excepts for profile
- Remove unused code

### Related Issues

- https://github.com/threefoldtech/tfgrid-sdk-ts/issues/1427

### Checklist

- [ ] Tests included
- [x] Build pass
- [ ] Documentation
- [x] Code format and docstrings
- [ ] Screenshots/Video attached (needed for UI changes)

### CODE REVIEW

File: packages/playground/src/components/manage_gateway_dialog.vue
Comment (0oM4R) [inline packages/playground/src/components/manage_gateway_dialog.vue:1]: do we need to use updateGrid here ? with the project name

Comment (zaelgohary) [reply packages/playground/src/components/manage_gateway_dialog.vue:1]: No we don't. Will remove it.

Comment (0oM4R) [reply packages/playground/src/components/manage_gateway_dialog.vue:1]: not sure about that; 
when i have one vm, the project name on manage gateway dialog is fullvm/vmName
image
but when i create another vm , the project name is only fullvm 
image

please have a look on this

Comment (0oM4R) [inline packages/playground/src/components/manage_gateway_dialog.vue:1]: i have 3 fullvms and tried to add domain to one of them, but the added domain appears on the three of them 
Screenshot from 2024-04-16 18-19-21
Screenshot from 2024-04-16 18-19-11
Screenshot from 2024-04-16 18-19-01

**Comment by zaelgohary** [comment]
Reverted changes in the deployment list as some issues appeared as a result of using the store. The previous issues were solved by reverting the changes.

**Review by 0oM4R — APPROVED** [review]
Great job!

**Comment by MohamedElmdary** [comment]
These parts are not yet updated
1. https://github.com/threefoldtech/tfgrid-sdk-ts/blob/52554262e4eaaf84f3af7c3055e2c59790b77df1/packages/playground/src/components/k8s_deployment_table.vue#L148

2. https://github.com/threefoldtech/tfgrid-sdk-ts/blob/52554262e4eaaf84f3af7c3055e2c59790b77df1/packages/playground/src/components/vm_deployment_table.vue#L162

3. https://github.com/threefoldtech/tfgrid-sdk-ts/blob/b2f0940803d2fde758be04980279419ae727a3f4/packages/playground/src/components/node_selector/TfNodeDetailsCard.vue#L244

4. https://github.com/threefoldtech/tfgrid-sdk-ts/blob/e3359fd6879ff0578247bb36ee09d1958209c2e0/packages/playground/src/components/ssh_keys/SshFormDialog.vue#L107

5. https://github.com/threefoldtech/tfgrid-sdk-ts/blob/6c98d0d08be55ecefb12c057bb917d04a156670b/packages/playground/src/dashboard/twin_view.vue#L192

**Comment by zaelgohary** [comment]
@MohamedElmdary Regarding the deployment files, previous comments noted that utilizing the store led to numerous bugs, so we've opted against using it in the listing. As for the remaining files, I'll make the required adjustments.

**Review by MohamedElmdary — APPROVED** [review]

**Pull request merged**